### PR TITLE
README.rst files for historians, aggregate historians, and tagging service

### PR DESCRIPTION
**Pull request by schandrika**
README.rst files for historians, aggregate historians, and tagging service. I haven't deleted the historian .rst files from docs folder though most of the contents of those files are now covered in the README. If the README.rst files look good we can delete the duplicate content in docs.
<a href='#crh-start'></a><a href='#crh-data-%7B%22processed%22%3A%20%5B%22https%3A//github.com/VOLTTRON/volttron/pull/1562%23discussion_r149510976%22%2C%20%22https%3A//github.com/VOLTTRON/volttron/pull/1562%23discussion_r150034910%22%2C%20%22https%3A//github.com/VOLTTRON/volttron/pull/1562%23discussion_r150034965%22%2C%20%22https%3A//github.com/VOLTTRON/volttron/pull/1562%23discussion_r150034989%22%2C%20%22https%3A//github.com/VOLTTRON/volttron/pull/1562%23discussion_r150035033%22%2C%20%22https%3A//github.com/VOLTTRON/volttron/pull/1562%23discussion_r150035069%22%2C%20%22https%3A//github.com/VOLTTRON/volttron/pull/1562%23discussion_r150035109%22%2C%20%22https%3A//github.com/VOLTTRON/volttron/pull/1562%23discussion_r150035135%22%2C%20%22https%3A//github.com/VOLTTRON/volttron/pull/1562%23discussion_r150035180%22%2C%20%22https%3A//github.com/VOLTTRON/volttron/pull/1562%23issuecomment-343233854%22%5D%2C%20%22comments%22%3A%20%7B%22Pull%206b3d7f230696b4e2607be6a49ce09b69be81cce0%20services/core/MongodbHistorian/README.rst%202%22%3A%20%7B%22html_url%22%3A%20%22https%3A//github.com/VOLTTRON/volttron/pull/1562%23discussion_r150035033%22%2C%20%22comments%22%3A%20%5B%7B%22body%22%3A%20%22Done%22%2C%20%22created_at%22%3A%20%222017-11-09T17%3A42%3A28Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars2.githubusercontent.com/u/16124920%3Fv%3D4%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/schandrika%22%7D%7D%5D%2C%20%22title%22%3A%20%22File%3A%20services/core/MongodbHistorian/README.rst%3AL1-182%22%7D%2C%20%22Pull%206b3d7f230696b4e2607be6a49ce09b69be81cce0%20services/core/SQLHistorian/README.rst%201%22%3A%20%7B%22html_url%22%3A%20%22https%3A//github.com/VOLTTRON/volttron/pull/1562%23discussion_r149510976%22%2C%20%22comments%22%3A%20%5B%7B%22body%22%3A%20%22Use%20_%20as%20separator%20as%20the%20others%20do%22%2C%20%22created_at%22%3A%20%222017-11-07T21%3A36%3A26Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars3.githubusercontent.com/u/3979063%3Fv%3D4%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/craig8%22%7D%7D%2C%20%7B%22body%22%3A%20%22Done%22%2C%20%22created_at%22%3A%20%222017-11-09T17%3A42%3A14Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars2.githubusercontent.com/u/16124920%3Fv%3D4%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/schandrika%22%7D%7D%5D%2C%20%22title%22%3A%20%22File%3A%20services/core/SQLHistorian/README.rst%3AL1-110%22%7D%2C%20%22Pull%206b3d7f230696b4e2607be6a49ce09b69be81cce0%20services/core/SQLHistorian/README.rst%2037%22%3A%20%7B%22html_url%22%3A%20%22https%3A//github.com/VOLTTRON/volttron/pull/1562%23discussion_r150034910%22%2C%20%22comments%22%3A%20%5B%7B%22body%22%3A%20%22fixed%22%2C%20%22created_at%22%3A%20%222017-11-09T17%3A42%3A01Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars2.githubusercontent.com/u/16124920%3Fv%3D4%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/schandrika%22%7D%7D%5D%2C%20%22title%22%3A%20%22File%3A%20services/core/SQLHistorian/README.rst%3AL1-110%22%7D%2C%20%22Pull%206b3d7f230696b4e2607be6a49ce09b69be81cce0%20services/core/CrateHistorian/README.rst%2040%22%3A%20%7B%22html_url%22%3A%20%22https%3A//github.com/VOLTTRON/volttron/pull/1562%23discussion_r150035135%22%2C%20%22comments%22%3A%20%5B%7B%22body%22%3A%20%22Done%22%2C%20%22created_at%22%3A%20%222017-11-09T17%3A42%3A53Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars2.githubusercontent.com/u/16124920%3Fv%3D4%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/schandrika%22%7D%7D%5D%2C%20%22title%22%3A%20%22File%3A%20services/core/CrateHistorian/README.rst%3AL1-79%22%7D%2C%20%22Pull%206b3d7f230696b4e2607be6a49ce09b69be81cce0%20services/core/CrateHistorian/README.rst%2065%22%3A%20%7B%22html_url%22%3A%20%22https%3A//github.com/VOLTTRON/volttron/pull/1562%23discussion_r150035109%22%2C%20%22comments%22%3A%20%5B%7B%22body%22%3A%20%22Done%22%2C%20%22created_at%22%3A%20%222017-11-09T17%3A42%3A47Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars2.githubusercontent.com/u/16124920%3Fv%3D4%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/schandrika%22%7D%7D%5D%2C%20%22title%22%3A%20%22File%3A%20services/core/CrateHistorian/README.rst%3AL1-79%22%7D%2C%20%22Pull%206b3d7f230696b4e2607be6a49ce09b69be81cce0%20services/core/MongodbAggregateHistorian/README.rst%206%22%3A%20%7B%22html_url%22%3A%20%22https%3A//github.com/VOLTTRON/volttron/pull/1562%23discussion_r150035069%22%2C%20%22comments%22%3A%20%5B%7B%22body%22%3A%20%22Done%22%2C%20%22created_at%22%3A%20%222017-11-09T17%3A42%3A37Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars2.githubusercontent.com/u/16124920%3Fv%3D4%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/schandrika%22%7D%7D%5D%2C%20%22title%22%3A%20%22File%3A%20services/core/MongodbAggregateHistorian/README.rst%3AL1-122%22%7D%2C%20%22Pull%206b3d7f230696b4e2607be6a49ce09b69be81cce0%20services/core/CrateHistorian/README.rst%202%22%3A%20%7B%22html_url%22%3A%20%22https%3A//github.com/VOLTTRON/volttron/pull/1562%23discussion_r150035180%22%2C%20%22comments%22%3A%20%5B%7B%22body%22%3A%20%22Done%22%2C%20%22created_at%22%3A%20%222017-11-09T17%3A43%3A00Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars2.githubusercontent.com/u/16124920%3Fv%3D4%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/schandrika%22%7D%7D%5D%2C%20%22title%22%3A%20%22File%3A%20services/core/CrateHistorian/README.rst%3AL1-79%22%7D%2C%20%22Pull%206b3d7f230696b4e2607be6a49ce09b69be81cce0%20services/core/SQLAggregateHistorian/config.mysql%201%22%3A%20%7B%22html_url%22%3A%20%22https%3A//github.com/VOLTTRON/volttron/pull/1562%23discussion_r150034989%22%2C%20%22comments%22%3A%20%5B%7B%22body%22%3A%20%22Done%22%2C%20%22created_at%22%3A%20%222017-11-09T17%3A42%3A21Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars2.githubusercontent.com/u/16124920%3Fv%3D4%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/schandrika%22%7D%7D%5D%2C%20%22title%22%3A%20%22File%3A%20services/core/SQLAggregateHistorian/config.mysql%3AL13-44%22%7D%2C%20%22General%20Comment%22%3A%20%7B%22html_url%22%3A%20%22https%3A//github.com/VOLTTRON/volttron/pull/1562%23issuecomment-343233854%22%2C%20%22comments%22%3A%20%5B%7B%22body%22%3A%20%22Fixed%20all%20review%20comments%22%2C%20%22created_at%22%3A%20%222017-11-09T17%3A43%3A23Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars2.githubusercontent.com/u/16124920%3Fv%3D4%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/schandrika%22%7D%7D%5D%2C%20%22title%22%3A%20%22General%20Comment%22%7D%7D%7D'></a>
<a href='https://www.codereviewhub.com/'></a>
- [x] <a href='#crh-comment-Pull 6b3d7f230696b4e2607be6a49ce09b69be81cce0 services/core/SQLHistorian/README.rst 1'></a> &nbsp;<b><a href='https://github.com/VOLTTRON/volttron/pull/1562#discussion_r149510976'>File: services/core/SQLHistorian/README.rst:L1-110</a></b>
- <a href='https://github.com/craig8'></a> Use _ as separator as the others do
- <a href='https://github.com/schandrika'></a> Done
- [x] <a href='#crh-comment-Pull 6b3d7f230696b4e2607be6a49ce09b69be81cce0 services/core/SQLHistorian/README.rst 37'></a> &nbsp;<b><a href='https://github.com/VOLTTRON/volttron/pull/1562#discussion_r150034910'>File: services/core/SQLHistorian/README.rst:L1-110</a></b>
- <a href='https://github.com/schandrika'></a> fixed
- [x] <a href='#crh-comment-Pull 6b3d7f230696b4e2607be6a49ce09b69be81cce0 services/core/SQLAggregateHistorian/config.mysql 1'></a> &nbsp;<b><a href='https://github.com/VOLTTRON/volttron/pull/1562#discussion_r150034989'>File: services/core/SQLAggregateHistorian/config.mysql:L13-44</a></b>
- <a href='https://github.com/schandrika'></a> Done
- [x] <a href='#crh-comment-Pull 6b3d7f230696b4e2607be6a49ce09b69be81cce0 services/core/MongodbHistorian/README.rst 2'></a> &nbsp;<b><a href='https://github.com/VOLTTRON/volttron/pull/1562#discussion_r150035033'>File: services/core/MongodbHistorian/README.rst:L1-182</a></b>
- <a href='https://github.com/schandrika'></a> Done
- [x] <a href='#crh-comment-Pull 6b3d7f230696b4e2607be6a49ce09b69be81cce0 services/core/MongodbAggregateHistorian/README.rst 6'></a> &nbsp;<b><a href='https://github.com/VOLTTRON/volttron/pull/1562#discussion_r150035069'>File: services/core/MongodbAggregateHistorian/README.rst:L1-122</a></b>
- <a href='https://github.com/schandrika'></a> Done
- [x] <a href='#crh-comment-Pull 6b3d7f230696b4e2607be6a49ce09b69be81cce0 services/core/CrateHistorian/README.rst 65'></a> &nbsp;<b><a href='https://github.com/VOLTTRON/volttron/pull/1562#discussion_r150035109'>File: services/core/CrateHistorian/README.rst:L1-79</a></b>
- <a href='https://github.com/schandrika'></a> Done
- [x] <a href='#crh-comment-Pull 6b3d7f230696b4e2607be6a49ce09b69be81cce0 services/core/CrateHistorian/README.rst 40'></a> &nbsp;<b><a href='https://github.com/VOLTTRON/volttron/pull/1562#discussion_r150035135'>File: services/core/CrateHistorian/README.rst:L1-79</a></b>
- <a href='https://github.com/schandrika'></a> Done
- [x] <a href='#crh-comment-Pull 6b3d7f230696b4e2607be6a49ce09b69be81cce0 services/core/CrateHistorian/README.rst 2'></a> &nbsp;<b><a href='https://github.com/VOLTTRON/volttron/pull/1562#discussion_r150035180'>File: services/core/CrateHistorian/README.rst:L1-79</a></b>
- <a href='https://github.com/schandrika'></a> Done
- [x] <a href='#crh-comment-General Comment'></a> &nbsp;<b><a href='https://github.com/VOLTTRON/volttron/pull/1562#issuecomment-343233854'>General Comment</a></b>
- <a href='https://github.com/schandrika'></a> Fixed all review comments

<a href='https://www.codereviewhub.com/VOLTTRON/volttron/pull/1562?mark_as_completed=1'></a>&nbsp;<a href='https://www.codereviewhub.com/VOLTTRON/volttron/pull/1562?approve=1'></a>&nbsp;<a href='https://github.com/VOLTTRON/volttron/pull/1562'></a>
<a href='#crh-end'></a>

### CODE REVIEW

**Comment by craig8** [comment]
This pull request is now managed with CodeReviewHub. <a href='https://github.com/VOLTTRON/volttron/pull/1562'>Refresh</a>.<a href='#crh-update-none'></a>

File: services/core/CrateHistorian/README.rst [hunk 1]
```diff
@@ -0,0 +1,78 @@
+.. _Crate-Historian:
+
```
Comment (craig8) [inline services/core/CrateHistorian/README.rst]: Please change to .. _Crate_Historian like the forwarder does

File: services/core/CrateHistorian/README.rst [hunk 2]
```diff
@@ -0,0 +1,78 @@
...
+    ::
+
+        sudo bash -c "$(curl -L install.crate.io)"
+
+This command will download and install all of the requirements for running
+crate, create a crate user and install a crate service.  After the installation
+the service will be available for viewing at http://localhost:4200 by default.
+
+.. note::  There is no authentication support within crate.
```
Comment (craig8) [inline services/core/CrateHistorian/README.rst]: Authentication for crate is an enterprise subscription only feature.

File: services/core/CrateHistorian/README.rst [hunk 3]
```diff
@@ -0,0 +1,78 @@
...
+
+        pip install crate
+
+
+Configuration
+~~~~~~~~~~~~~
+
+Because there is no authorization to access a crate database the configuration
+for the CrateHistorian is very easy.
+
```
Comment (craig8) [inline services/core/CrateHistorian/README.rst]: The following is an example of the crate historian's configuration.

File: services/core/MongodbAggregateHistorian/README.rst [hunk 4]
```diff
@@ -0,0 +1,121 @@
+.. _SQL_Aggregate_Historian:
+
+=======================
+SQL Aggregate Historian
+=======================
+
```
Comment (craig8) [inline services/core/MongodbAggregateHistorian/README.rst]: Mongodb Aggregate Historian?

File: services/core/MongodbHistorian/README.rst [hunk 5]
```diff
@@ -0,0 +1,181 @@
+.. _Mongo-Historian:
+
```
Comment (craig8) [inline services/core/MongodbHistorian/README.rst]: Use underscore in the link

File: services/core/SQLAggregateHistorian/config.mysql [hunk 6]
```diff
@@ -13,36 +13,31 @@
     # configuration from mysql historian - END
```
Comment (craig8) [inline services/core/SQLAggregateHistorian/config.mysql]: Should only be one config file inside the SQLAggregateHistorian directory.  Note the configuration should be setup to use one or the other and have directions in the readme to how to change it.

File: services/core/SQLHistorian/README.rst [hunk 7]
```diff
@@ -0,0 +1,109 @@
+.. _SQL-Historian:
```
Comment (craig8) [inline services/core/SQLHistorian/README.rst]: Use _ as separator as the others do

File: services/core/SQLHistorian/README.rst [hunk 8]
```diff
@@ -0,0 +1,109 @@
...
+3. The tables in the sql database be created as part of the execution of
+   the SQLHistorianAgent only if the database user has CREATE privileges.
+   If not, use the mysql-create*.sql script to create the tables and then
+   start agent.
+
+Dependencies
+------------
+
+In order to use mysql one must install the **mysql-python connector**
+
```
Comment (craig8) [inline services/core/SQLHistorian/README.rst]: Lets get rid of thee first one as it never works and just go with the
pip install mysql-connector-python-rf

Comment (schandrika) [reply services/core/SQLHistorian/README.rst]: fixed

File: services/core/SQLHistorian/README.rst (hunk 7 above)
Comment (schandrika) [reply services/core/SQLHistorian/README.rst]: Done

File: services/core/SQLAggregateHistorian/config.mysql (hunk 6 above)
Comment (schandrika) [reply services/core/SQLAggregateHistorian/config.mysql]: Done

File: services/core/MongodbHistorian/README.rst (hunk 5 above)
Comment (schandrika) [reply services/core/MongodbHistorian/README.rst]: Done

File: services/core/MongodbAggregateHistorian/README.rst (hunk 4 above)
Comment (schandrika) [reply services/core/MongodbAggregateHistorian/README.rst]: Done

File: services/core/CrateHistorian/README.rst (hunk 3 above)
Comment (schandrika) [reply services/core/CrateHistorian/README.rst]: Done

File: services/core/CrateHistorian/README.rst (hunk 2 above)
Comment (schandrika) [reply services/core/CrateHistorian/README.rst]: Done

File: services/core/CrateHistorian/README.rst (hunk 1 above)
Comment (schandrika) [reply services/core/CrateHistorian/README.rst]: Done

**Comment by schandrika** [comment]
Fixed all review comments

**Pull request merged**